src/test/javascript/spec/component/exam/exam-timer.component.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ArtemisTestModule } from '../../test.module';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { ExamTimerComponent } from 'app/exam/participate/timer/exam-timer.component';
+import { ArtemisServerDateService } from 'app/shared/server-date.service';
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import * as moment from 'moment';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { MockRouter } from '../../helpers/mocks/service/mock-route.service';
+import { TranslatePipe, TranslateService } from '@ngx-translate/core';
+import { Router } from '@angular/router';
+import { MockPipe } from 'ng-mocks';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('ExamTimerComponent', function () {
+    let component: ExamTimerComponent;
+    let fixture: ComponentFixture<ExamTimerComponent>;
+    let dateService: ArtemisServerDateService;
+
+    const now = moment();
+    const inFuture = moment().add(100, 'ms');
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [ExamTimerComponent, MockPipe(TranslatePipe)],
+            providers: [
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: Router, useValue: MockRouter },
+            ],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ExamTimerComponent);
+        component = fixture.componentInstance;
+        dateService = TestBed.inject(ArtemisServerDateService);
+        component.endDate = inFuture;
+    });
+
+    it('should call ngOnInit', () => {
+        spyOn(dateService, 'now').and.returnValue(now);
+        component.criticalTime = moment.duration(200);
+        expect(component.endDate).to.equal(inFuture);
+        component.ngOnInit();
+        expect(component).to.be.ok;
+        expect(component.isCriticalTime).to.be.true;
+    });
+
+    it('should update display times', () => {
```

Comment:
I added a test in 486fa378fe75ae455478b8f1a15d009470622a35 that also tests the async behaviour.

Reply:
Sure, test looks good :) And also covers an aspect i did not quite catch in my tests